packages/protocol/contracts/L1/verifiers/PseZkVerifier.sol

```diff
@@ -29,68 +44,106 @@ contract PseZkVerifier is EssentialContract, IVerifier {
-        if (isContesting) return;
-
-        bytes32 instance = calcInstance(prover, evidence);
+        if (input.isContesting) return;
+
+        PseZkEvmProof memory proof = abi.decode(evidence.proof, (PseZkEvmProof));
+
+        bytes32 instance;
+        if (input.blobUsed) {
+            PointProof memory pf = abi.decode(proof.pointProof, (PointProof));
+
+            instance = calcInstance({
+                prover: input.prover,
+                evidence: evidence,
+                pointValue: pf.pointValue,
+                metaHash: input.metaHash
+            });
+
+            Lib4844.evaluatePoint({
+                blobHash: input.blobHash,
+                x: calc4844PointEvalX(input.blobHash, pf.txListHash),
+                y: pf.pointValue,
+                commitment: pf.pointCommitment,
+                proof: pf.pointProof
+            });
+        } else {
+            assert(proof.pointProof.length == 0);
+            instance = calcInstance({
+                prover: input.prover,
+                evidence: evidence,
+                pointValue: 0,
+                metaHash: input.metaHash
+            });
+        }
 
         // Validate the instance using bytes utilities.
         bool verified = LibBytesUtils.equal(
-            LibBytesUtils.slice(evidence.proof, 2, 32),
+            LibBytesUtils.slice(proof.zkp, 0, 32),
             bytes.concat(bytes16(0), bytes16(instance))
         );
+
         if (!verified) revert L1_INVALID_PROOF();
 
         verified = LibBytesUtils.equal(
-            LibBytesUtils.slice(evidence.proof, 34, 32),
+            LibBytesUtils.slice(proof.zkp, 32, 32),
             bytes.concat(bytes16(0), bytes16(uint128(uint256(instance))))
         );
         if (!verified) revert L1_INVALID_PROOF();
 
-        // Extract verifier ID from the proof.
-        uint16 verifierId = uint16(bytes2(evidence.proof[0:2]));
-
         // Delegate to the ZKP verifier library to validate the proof.
         // Resolve the verifier's name and obtain its address.
-        address verifierAddress = resolve(getVerifierName(verifierId), false);
+        address verifierAddress =
+            resolve(getVerifierName(proof.verifierId), false);
 
         // Call the verifier contract with the provided proof.
         bytes memory ret;
-        (verified, ret) =
-            verifierAddress.staticcall(bytes.concat(evidence.proof[2:]));
+        (verified, ret) = verifierAddress.staticcall(bytes.concat(proof.zkp));
 
         // Check if the proof is valid.
         if (!verified) revert L1_INVALID_PROOF();
         if (ret.length != 32) revert L1_INVALID_PROOF();
         if (bytes32(ret) != keccak256("taiko")) revert L1_INVALID_PROOF();
     }
 
+    function calc4844PointEvalX(
+        bytes32 blobHash,
+        bytes32 txListHash
+    )
+        public
+        pure
+        returns (uint256)
+    {
+        return uint256(blobHash ^ txListHash) % Lib4844.BLS_MODULUS;
```

Comment:
@Brechtpd Do you think this will work as well, as an cheaper alternative to keccak?

Reply:
Would keep the keccak method because it's easier to do in the circuit and it only needs to be done once. Doing the xor also requires quite a lot of lookups, and we don't have a dedicated circuit to do that specifically like we do with keccak.

Reply:
fixed